tests/integration/test_logcollector/test_keep_running/test_keep_running.py

```diff
@@ -0,0 +1,124 @@
+        get_local_internal_options (fixture): Get internal configuration.
+        configure_local_internal_options (fixture): Set internal configuration for testing.
+        get_configuration (fixture): Get configurations from the module.
+        configure_environment (fixture): Configure a custom environment for testing.
+        generate_log_file (fixture): Generate a log file for testing.
+        restart_logcollector (fixture): Reset log file and start a new monitor.
+    """
+    config = get_configuration['metadata']
+
+    # Add first line to log
+    message = config['log_line_before']
+    with open(config['location'], 'a') as f:
+        f.write(f"{message}\n")
+
+    # Ensure that the file is being analyzed
+    message = fr"INFO: \(\d*\): Analyzing file: '{log_test_path}'."
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
+
+    # Add another line to log
+    message = config['log_line_before']
+    with open(config['location'], 'a') as f:
+        f.write(f"{message}\n")
+
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
+
+    file.truncate_file(config['location'])
+
+    # Ensure that the rotation has been completed:
+    message = f"DEBUG: File size reduced. {log_test_path}"
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
+    # Add first line to rotated log
+    message = config['log_line_after']
+    with open(config['location'], 'a') as f:
+        f.write(f"{message}\n")
+
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
```

Comment:
```suggestion
    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
                            callback=callback_message)
```

Reply:
Done in: https://github.com/wazuh/wazuh-qa/pull/1300/commits/a01dc4462069a2ff448c4d9538d4d38741fe9328